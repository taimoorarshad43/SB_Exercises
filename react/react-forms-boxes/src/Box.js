import React from "react";

function Box({
  id,
  handleRemove,
  width = 5,
  height = 5,
  backgroundColor = "blue"
}) {
  const remove = () => handleRemove(id);
  
  return (
    <div style={{ display: 'flex', alignItems: 'center', margin: '5px' }}>
      <div
        style={{
          height: `${height}em`,
          width: `${width}em`,
          backgroundColor,
          border: '1px solid #ccc',
          marginRight: '5px'
        }}
      />
      <button onClick={remove} style={{ padding: '2px 6px' }}>X</button>
    </div>
  );
}

export default Box;
