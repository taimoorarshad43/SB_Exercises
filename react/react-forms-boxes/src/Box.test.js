import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

it("renders without crashing", function() {
  render(<Box />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});

it("renders with custom props", function() {
  const { asFragment } = render(
    <Box 
      width={10} 
      height={15} 
      backgroundColor="red" 
      id="test-id" 
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it("calls handleRemove when X button is clicked", function() {
  const removeMock = jest.fn();
  const { getByText } = render(
    <Box 
      id="test-id" 
      handleRemove={removeMock} 
    />
  );
  const removeButton = getByText("X");
  fireEvent.click(removeButton);
  expect(removeMock).toHaveBeenCalledWith("test-id");
});
