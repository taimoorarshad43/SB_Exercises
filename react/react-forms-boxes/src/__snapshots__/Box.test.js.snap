// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: flex; align-items: center; margin: 5px;"
  >
    <div
      style="height: 5em; width: 5em; background-color: blue; border: 1px solid #ccc; margin-right: 5px;"
    />
    <button
      style="padding: 2px 6px;"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;

exports[`renders with custom props 1`] = `
<DocumentFragment>
  <div
    style="display: flex; align-items: center; margin: 5px;"
  >
    <div
      style="height: 15em; width: 10em; background-color: red; border: 1px solid #ccc; margin-right: 5px;"
    />
    <button
      style="padding: 2px 6px;"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
